extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetHadoopInputRowParser.java

```diff
@@ -56,15 +63,41 @@ public ParquetHadoopInputRowParser(
     }
   }
 
+  @Nullable
+  private LogicalType determineTimestampSpecLogicalType(Schema schema, String timestampSpecField)
+  {
+    for (Schema.Field field : schema.getFields()) {
+      if (field.name().equals(timestampSpecField)) {
+        return field.schema().getLogicalType();
+      }
+    }
+    return null;
+  }
+
   /**
    * imitate avro extension {@link AvroStreamInputRowParser#parseGenericRecord(GenericRecord, ParseSpec, List, boolean, boolean)}
    */
   @Override
   public InputRow parse(GenericRecord record)
   {
+    // Map the record to a map
     GenericRecordAsMap genericRecordAsMap = new GenericRecordAsMap(record, false, binaryAsString);
-    TimestampSpec timestampSpec = parseSpec.getTimestampSpec();
-    DateTime dateTime = timestampSpec.extractTimestamp(genericRecordAsMap);
+
+    // Determine logical type of the timestamp column
+    LogicalType logicalType = determineTimestampSpecLogicalType(record.getSchema(), timestampSpec.getTimestampColumn());
+
+    // Parse time timestamp based on the parquet schema.
+    // https://github.com/Parquet/parquet-format/blob/master/LogicalTypes.md#date
```

Comment:
The markdown file will probably not change a lot, but good point. I've updated it to the latest commit.